packages/ui-extensions-server-kit/src/ExtensionServerClient/ExtensionServerClient.ts

```diff
@@ -140,6 +179,35 @@ export class ExtensionServerClient implements ExtensionServer.Client {
       this.connection?.close()
     }
   }
+
+  private _getLocalizedExtensions(extensions?: ExtensionPayload[]) {
+    return extensions?.map((extension) => {
+      if (!this.options.locales) {
+        return extension
+      }
+
+      const shouldUpdateTranslations =
+        this.extensionsByUuid[extension.uuid] ||
+        this.extensionsByUuid[extension.uuid]?.localization?.lastUpdated !== extension.localization?.lastUpdated
+
+      const localization = shouldUpdateTranslations
+        ? getFlattenedLocalization(extension.localization, this.options.locales)
+        : this.extensionsByUuid[extension.uuid].localization
+
+      this.extensionsByUuid[extension.uuid] = {
+        ...extension,
```

Comment:
No, I don't think it should as currently the `lastUpdated` refers to the last time the Developer has made changes to the localization. This part of the code is called every time an update message is sent through the websocket and just modifies the payload so we get flattened translations instead of the raw values.